xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authentication/xwiki-platform-security-authentication-api/src/main/java/org/xwiki/security/authentication/api/ResetPasswordManager.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.security.authentication.api;
```

Comment:
Why ".api" ? Everything which is not internal is an API.

Reply:
So sadly that's not something I introduced here, but I introduced this package in 11.6 when working on the authentication failure manager, see: https://github.com/xwiki/xwiki-platform/tree/master/xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authentication/xwiki-platform-security-authentication-api/src/main/java/org/xwiki/security/authentication/api. 
We can change it now but it would be breaking... I felt that it would be cleaner to have all our APIs in the same package, even if the name is wrong.

Reply:
I think we should fix it now. Cannot we keep the .api one in legacy for example?

In any case I think it's better for the future to have a clean api , especially as it's growing. 

@tmortagne  WDYT?

Reply:
New API like ResetPasswordManager should definitely not go in this bad package IMO. The existing ones also don't really have much to do with ResetPasswordManager except for the fact that they are generally about authentication. There is no rule that say all APIs of a modules should go in the same package and I would really not see the point.

Now for what to do about the existing ones it does not have anything to do with this PR, but I suggest deprecating them and copy them in a more accurate package.

Reply:
> New API like ResetPasswordManager should definitely not go in this bad package IMO.

Ok I'll fix this one here. 

> Now for what to do about the existing ones it does not have anything to do with this PR, but I suggest deprecating them and copy them in a more accurate package.

Agreed that it shouldn't be part of this PR, I'll open a ticket for it to not forget.